### PR TITLE
fix: adjust RsPopulation class to be compatible with Iterator class

### LINKED ISSUE #6275: Automated Measures report PHP 8.2 deprecated message
Automated Measures report PHP 8.2 deprecation message (Report-> Clinic -> Automated Measures)
Here is message in error log:
```
NOTICE: PHP message: PHP Deprecated:  Return type of RsPopulation::current() should either be compatible with Iterator::current(): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /usr/share/nginx/html/openemr/library/classes/rulesets/library/RsPopulation.php on line 47
NOTICE: PHP message: PHP Deprecated:  Return type of RsPopulation::next() should either be compatible with Iterator::next(): void, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /usr/share/nginx/html/openemr/library/classes/rulesets/library/RsPopulation.php on line 57
NOTICE: PHP message: PHP Deprecated:  Return type of RsPopulation::key() should either be compatible with Iterator::key(): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /usr/share/nginx/html/openemr/library/classes/rulesets/library/RsPopulation.php on line 52
NOTICE: PHP message: PHP Deprecated:  Return type of RsPopulation::offsetGet($offset) should either be compatible with ArrayAccess::offsetGet(mixed $offset): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /usr/share/nginx/html/openemr/library/classes/rulesets/library/RsPopulation.php on line 94
```

### PR DESCRIPTION
**Pull request by stephenwaite**
Fixes #6275

### CODE REVIEW

**Comment by stephenwaite** [comment]
used https://www.php.net/manual/en/class.iterator.php#iterator.example.basic as reference

File: library/classes/rulesets/library/RsPopulation.php [hunk 1]
```diff
@@ -39,29 +42,30 @@ public function count(): int
...
     {
-        return key($this->_patients);
+        return $this->position;
     }
 
-    public function next()
+    public function next(): void
     {
-        return next($this->_patients);
+        ++$this->position;
```
Comment (bradymiller) [inline library/classes/rulesets/library/RsPopulation.php:63]: is this supposed to return the patient (value) or the position (key)?

Comment (bradymiller) [reply library/classes/rulesets/library/RsPopulation.php:63]: ow wait, disregard. it is not returning anything :)

**Review by bradymiller — APPROVED** [review]

**Comment by bradymiller** [comment]
If testing well, then bring it on in :)

dance-bear

**Comment by stephenwaite** [comment]
testing well, bringing in :crossed_fingers:

**Pull request merged**